server/engine/src/main/java/org/eclipse/lsp/cobol/service/delegates/completions/Keywords.java

```diff
@@ -14,39 +14,77 @@
  */
 package org.eclipse.lsp.cobol.service.delegates.completions;
 
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
 import com.google.inject.Singleton;
+import lombok.extern.slf4j.Slf4j;
 import org.eclipse.lsp.cobol.core.engine.dialects.daco.DaCoDialect;
-import org.eclipse.lsp.cobol.core.engine.dialects.idms.IdmsDialect;
+import org.eclipse.lsp.cobol.service.SettingsService;
 
+import java.io.IOException;
 import java.io.InputStream;
 import java.io.SequenceInputStream;
 import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
 
 /** This class is a provider for Cobol keywords and their descriptions */
 @Singleton
-public class Keywords extends CompletionStorage {
+@Slf4j
+public class Keywords extends CompletionStorage<String> {
   private static final String KEYWORDS_FILE_PATH = "LanguageKeywords.txt";
   private static final String KEYWORDS_IDMS_FILE_PATH = "LanguageKeywordsIDMS.txt";
   private static final String KEYWORDS_DACO_FILE_PATH = "LanguageKeywordsDaCo.txt";
 
+  @Inject
+  Keywords(SettingsService settingsService) {
+    super(settingsService);
+  }
+
   @Override
-  protected InputStream getInputStream(List<String> dialectList) {
-    if (!dialectList.isEmpty()
-        && dialectList.contains(IdmsDialect.NAME)
-        && dialectList.contains(DaCoDialect.NAME)) {
-      return new SequenceInputStream(
-          Keywords.class.getResourceAsStream(KEYWORDS_FILE_PATH),
-          new SequenceInputStream(
-              Keywords.class.getResourceAsStream(KEYWORDS_IDMS_FILE_PATH),
-              Keywords.class.getResourceAsStream(KEYWORDS_DACO_FILE_PATH)));
-    } else if (!dialectList.isEmpty() && dialectList.contains(IdmsDialect.NAME)) {
-      return new SequenceInputStream(
-          Keywords.class.getResourceAsStream(KEYWORDS_FILE_PATH),
-          Keywords.class.getResourceAsStream(KEYWORDS_IDMS_FILE_PATH));
-    } else if (!dialectList.isEmpty() && dialectList.contains(DaCoDialect.NAME)) {
-      return new SequenceInputStream(
-          Keywords.class.getResourceAsStream(KEYWORDS_FILE_PATH),
-          Keywords.class.getResourceAsStream(KEYWORDS_DACO_FILE_PATH));
-    } else return Keywords.class.getResourceAsStream(KEYWORDS_FILE_PATH);
+  protected Map<String, String> getInputStream(List<String> dialectList) {
+    Properties props = new Properties();
+    InputStream stream;
+    if (!dialectList.isEmpty()) {
+      if (dialectList.size() == 2
+              || (dialectList.size() == 1 && dialectList.contains(DaCoDialect.NAME))) {
+        stream =
```

Comment:
Let's move all this block to some method that returns a `stream` based on dialectList, for example. It will be easier to read it and test it